Range test failure for complex objects

## Expected Behavior
Green.

## Current Behavior
https://rest.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/1634-1715615921/index.html#suites/87d8da5f92824d091055170c29079bc0/3a1cec7d64a7a14/
https://rest.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/382-1715098385/index.html
 tests.object.test_object_api.TestObjectApi#test_object_get_range and  tests.object.test_object_api.TestObjectApi#test_object_get_range_hash fail in various ways:
```
AssertionError: Expected range hash to match 524288:524288 slice of file payload
assert '676ef3151297...bc0f60358f433' == 'e3b0c44298fc...5991b7852b855'
  - e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
  + 676ef31512976a04c097d6c022162e01d6cdd4286e6730968e5bc0f60358f433
```
```
Command: neofs-cli --config /home/runner/work/neofs-node/neofs-node/neofs-testcases/wallet_config.yml object range --rpc-endpoint 'localhost:59125' --wallet '/home/runner/work/neofs-node/neofs-node/neofs-testcases/TemporaryDir/9f834a91-57b4-4082-9bd9-79bc82ca9a77.json' --cid '6VNpofYWp8Sxh3WBJv2swxtdXe8UGcdT6L6GKsxmNZH8' --oid 'BiYm5UuS4LmaPWVKeXARcMrPaRCeyRddhUHMabrpMqqr' --range '524288:524288' --file 'TemporaryDir/TestObjectsDir/0a7ef3e6-611a-45cd-9ec5-cc92f3b4f753'
Error:
return code: 1
output: can't get object payload range: copy payload: unexpected EOF
```
```
RuntimeError: Command: neofs-cli --config /home/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/wallet_config.yml object hash --rpc-endpoint 'localhost:58891' --wallet '/home/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/TemporaryDir/7c89ec08-da9e-43d8-abdb-df524dbe2a2e.json' --cid '5yeceaZvNT7a9enMfpcZsVsCpbShwVQmd9JTYGJUEVD7' --oid '68ik5gVENPhaC3AzePAeZBV4jJqmueSVZKDLhcbX799P' --range '524288:524288'
Error:
return code: 1
output: rpc error: read payload hashes via client: status: code = 2049 message = object not found
```

## Possible Solution
Fix the damn thing.

## Steps to Reproduce (for bugs)
Run neofs-testcases.

## Your Environment
* Version used: current master, 0.41.1+

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ Changelog for NeoFS Node
 ### Added
 
 ### Fixed
+- GETRANGE request may fail in certain cases (#2849)
 
 ### Changed
 - SN API server responds with status message even to old clients from now (#2846)

--- a/pkg/services/object/get/assembly_v2.go
+++ b/pkg/services/object/get/assembly_v2.go
@@ -156,7 +156,7 @@ func requiredChildren(rng *objectSDK.Range, children []objectSDK.MeasuredObject)
 		size := uint64(child.ObjectSize())
 		bytesSeen += size
 
-		if bytesSeen < leftBound {
+		if bytesSeen <= leftBound {
 			continue
 		}
 

--- a/pkg/services/object/get/assembly_v2_test.go
+++ b/pkg/services/object/get/assembly_v2_test.go
@@ -77,6 +77,10 @@ func Test_RequiredChildren(t *testing.T) {
 					rightBound = lastChildBound
 				}
 
+				if len(payloads[i]) > 0 {
+					require.NotZero(t, rightBound-leftBound, "do not ask for empty payload ever")
+				}
+
 				payloadFromChild := payloads[i][leftBound:rightBound]
 				res = append(res, payloadFromChild...)
 			}

